### ISSUE TITLE
BatchNormalization:parameters() not properly implemented

### ISSUE THREAD

**Issue by chtlp** [torch/nn#632]
The default implementation of `Module:parameters()` only considers `weight` and `bias`, while `BatchNormalization` and `SptialBatchNormalization` have additional parameters, e.g. `running_mean` and `running_var` or `running_std`. I think the following code could work, but can anyone confirm this issue and add this simple fix?

```lua
function nn.SpatialBatchNormalization:parameters()
    local params = {}
    if self.weight then
        params[#params+1] = self.weight
    end
    if self.bias then
        params[#params+1] = self.bias
    end
    params[#params+1] = self.running_mean
    params[#params+1] = self.running_var
    return params, {}
end

function nn.BatchNormalization:parameters()
    return {self.weight, self.bias,
        self.running_mean, self.running_std}, {}
end

```

**Comment by chtlp** [comment]
It turns out this would disrupt the optimization algorithm because these parameters can't be updated by gradient descent. I wanted to save all parameters of a model via `Module:parameters`, but it seems to be a problematic approach.
